lib/github/github/file.ex

```diff
@@ -0,0 +1,41 @@
+defmodule BorsNG.GitHub.File do
+  @moduledoc """
+  The structure of GitHub pull request file
+  """
+
+  @type tjson :: map
+  @type t :: %BorsNG.GitHub.File{
+          filename: bitstring
+        }
+  defstruct(filename: "")
+
+  @doc """
+  Convert from Poison-decoded JSON to a Pr struct.
+  """
+  @spec from_json!(tjson) :: [t]
+  def from_json!(json) do
+    {:ok, files} = from_json(json)
+    files
+  end
+
+  @doc """
+  Convert from Poison-decoded JSON to a Pr struct.
+  """
+  @spec from_json(tjson) :: {:ok, [t]} | {:error}
```

Comment:
you forgot the second element returned on `{:error}`, it should be something like `from_json(input) :: {ok, [t]} | {:error, input} when input: map`

Reply:
⚡️